segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/IndexAppendProcessor.java

```diff
@@ -64,28 +86,39 @@ protected void processAppend(String segmentName, long indexSegmentEventSize) {
                     getIndexSegmentName(segmentName), indexSegmentEventSize, NameUtils.INDEX_APPEND_EVENT_SIZE);
             return;
         }
-        appendProcessor.updateItem(segmentName, indexSegmentEventSize);
+        appendProcessor.process(new QueuedSegment(segmentName));
     }
 
-    private void handleIndexAppend(String segmentName, long indexSegmentEventSize) {
-        store.getStreamSegmentInfo(segmentName, TIMEOUT)
-                .thenCompose(info -> {
-                    long eventCount = info.getAttributes().get(EVENT_COUNT) != null ? info.getAttributes().get(EVENT_COUNT) : 0;
-                    ByteBufWrapper byteBuff = getIndexAppendBuf(info.getLength(), eventCount);
-                    AttributeUpdateCollection attributes = AttributeUpdateCollection.from(
-                            new AttributeUpdate(EVENT_COUNT, AttributeUpdateType.ReplaceIfGreater, eventCount));
-                    return store.append(getIndexSegmentName(segmentName), byteBuff, attributes, TIMEOUT);
-
-                }).thenAccept(v -> log.trace("Index segment append successful for segment {} ", getIndexSegmentName(segmentName)))
-                .exceptionally(ex -> {
-                    log.warn("Index segment append failed for segment {} due to ", getIndexSegmentName(segmentName), ex);
-                    return null;
-                });
+    private CompletableFuture<Void> handleIndexAppend(QueuedSegment segment) {
+        return store.getStreamSegmentInfo(segment.segmentName, TIMEOUT).thenCompose(info -> {
+            long eventCount = info.getAttributes().get(EVENT_COUNT) != null ? info.getAttributes().get(EVENT_COUNT) : 0;
+            ByteBufWrapper byteBuff = getIndexAppendBuf(info.getLength(), eventCount);
+            AttributeUpdateCollection attributes = AttributeUpdateCollection.from(
+                new AttributeUpdate(EVENT_COUNT, AttributeUpdateType.ReplaceIfGreater, eventCount));
+            return store.append(getIndexSegmentName(segment.segmentName), byteBuff, attributes, TIMEOUT);
+        }).thenRun(() -> {
+            log.trace("Index segment append successful for segment {} ", getIndexSegmentName(segment.segmentName));
+        }).exceptionally(ex -> {
+            log.warn(
+                "Index segment append failed for segment {} due to ", getIndexSegmentName(segment.segmentName), ex);
+            return null;
+        });
     }
 
     private ByteBufWrapper getIndexAppendBuf(Long eventLength, Long eventCount) {
         IndexEntry indexEntry = new IndexEntry(eventLength, eventCount, System.currentTimeMillis());
         return new ByteBufWrapper(Unpooled.wrappedBuffer( indexEntry.toBytes().getCopy()));
     }
 
+    @Override
+    public void close() {
+        appendProcessor.shutdown();
+    }
+    
+    @VisibleForTesting
+    void runRemainingAndClose() {
```

Comment:
This method is not called from anywhere.

Reply:
It is being called from `AppendProcessorTest` and `IndexAppendProcessorTest`